eng/packages/http-client-csharp-mgmt/generator/TestProjects/Local/Mgmt-TypeSpec/src/Generated/FooCollection.cs

```diff
@@ -46,24 +48,300 @@ internal static void ValidateResourceId(ResourceIdentifier id)
             }
         }
 
-        /// <summary> List Foo resources by resource group. </summary>
+        /// <summary> Create a Foo. </summary>
+        /// <param name="waitUntil"> <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service; <see cref="WaitUntil.Started"/> if it should return after starting the operation. For more information on long-running operations, please see <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md"> Azure.Core Long-Running Operation samples</see>. </param>
+        /// <param name="fooName"> The name of the Foo. </param>
+        /// <param name="resource"> Resource create parameters. </param>
         /// <param name="cancellationToken"> The cancellation token that can be used to cancel the operation. </param>
-        public virtual Pageable<FooResource> GetAll(CancellationToken cancellationToken = default) => throw new NotImplementedException();
+        /// <exception cref="ArgumentNullException"> <paramref name="fooName"/> or <paramref name="resource"/> is null. </exception>
+        public virtual async Task<ArmOperation<FooResource>> CreateOrUpdateAsync(WaitUntil waitUntil, string fooName, FooData resource, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(fooName, nameof(fooName));
+            Argument.AssertNotNull(resource, nameof(resource));
+
+            using DiagnosticScope scope = _fooClientDiagnostics.CreateScope("FooCollection.CreateOrUpdateAsync");
+            scope.Start();
+            try
+            {
+                RequestContext context = new RequestContext
+                {
+                    CancellationToken = cancellationToken
+                }
+                ;
```

Comment:
this looks like a bug in MTG if we have built our expressions properly.
Could we create an issue to track this?

Reply:
an issue https://github.com/microsoft/typespec/issues/7283 has been created by @ArcturusZhang , thanks.